File: experiments/puppeteer-automation/saucedemo-e2e/tests/cart.test.ts
import assert from 'node:assert/strict';
import { describe, it } from 'node:test';
import { CartPage, InventoryPage } from '../src/pages/saucedemo.pages.js';
import { PRODUCTS } from '../src/utils/test-data.js';
import { withAuthenticatedPage } from './harness.js';

describe('Cart', () => {
  it('displays added items', async () => {
    await withAuthenticatedPage('cart-display', async (page) => {
      const inventory = new InventoryPage(page);
      await inventory.addToCart(PRODUCTS.backpack);
      await inventory.addToCart(PRODUCTS.boltTshirt);
      await inventory.openCart();

      const cart = new CartPage(page);
      await cart.waitForLoaded();
      assert.equal(await cart.hasItem(PRODUCTS.backpack), true);
      assert.equal(await cart.hasItem(PRODUCTS.boltTshirt), true);
    });
  });

  it('removes item and updates badge', async () => {
    await withAuthenticatedPage('cart-remove', async (page) => {
      const inventory = new InventoryPage(page);
      await inventory.addToCart(PRODUCTS.backpack);
      await inventory.addToCart(PRODUCTS.boltTshirt);
      await inventory.openCart();

      const cart = new CartPage(page);
      await cart.removeItem(PRODUCTS.backpack);
      assert.equal(await cart.hasItem(PRODUCTS.backpack), false);

      await cart.continueShopping();
      await inventory.waitForLoaded();
      assert.equal(await inventory.getCartBadgeText(), '1');
    });
  });

  it('navigates to checkout', async () => {
    await withAuthenticatedPage('cart-checkout', async (page) => {
      const inventory = new InventoryPage(page);
      await inventory.addToCart(PRODUCTS.backpack);
      await inventory.openCart();

      const cart = new CartPage(page);
      await cart.checkout();
      assert.match(page.url(), /checkout-step-one/);
    });
  });
});

File: experiments/selenium-java/saucedemo-e2e/src/test/java/com/qaskills/saucedemo/tests/CartTest.java
package com.qaskills.saucedemo.tests;

import com.qaskills.saucedemo.pages.CartPage;
import com.qaskills.saucedemo.pages.CheckoutPage;
import com.qaskills.saucedemo.pages.InventoryPage;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class CartTest extends BaseTest {

    private static final String BACKPACK = "Sauce Labs Backpack";
    private static final String BOLT_TSHIRT = "Sauce Labs Bolt T-Shirt";

    private CartPage cartPage;

    @BeforeMethod(dependsOnMethods = "setUp")
    public void prepareCart() {
        InventoryPage inventoryPage = loginAsStandardUser();
        inventoryPage.addProductToCart(BACKPACK);
        inventoryPage.addProductToCart(BOLT_TSHIRT);
        cartPage = inventoryPage.openCart();
        cartPage.waitForPage();
    }

    @Test(description = "Verify items appear in cart")
    public void testCartDisplaysItems() {
        assertThat(cartPage.isItemInCart(BACKPACK)).isTrue();
        assertThat(cartPage.isItemInCart(BOLT_TSHIRT)).isTrue();
    }

    @Test(description = "Verify remove item from cart")
    public void testRemoveItemFromCart() {
        cartPage.removeItem(BACKPACK);

        assertThat(cartPage.isItemInCart(BACKPACK)).isFalse();
        InventoryPage inventoryPage = cartPage.continueShopping();
        inventoryPage.waitForPage();
        assertThat(inventoryPage.getCartBadgeText()).isEqualTo("1");
    }

    @Test(description = "Verify navigate to checkout")
    public void testNavigateToCheckout() {
        CheckoutPage checkoutPage = cartPage.proceedToCheckout();
        checkoutPage.waitForStepOne();
        assertThat(checkoutPage.getCurrentUrl()).contains("checkout-step-one.html");
    }
}
